Mathlib/RingTheory/Localization/FractionRingAlgebra.lean

```diff
@@ -0,0 +1,35 @@
+import Mathlib.RingTheory.Localization.Integral
+import Mathlib.RingTheory.RingHom.Finite
+
+/-! # Instances about extensions of fraction rings -/
+
+variable {R₀ R S : Type*} [CommRing R] [CommRing S] [Algebra R S] [FaithfulSMul R S]
+  [IsDomain R] [IsDomain S] [CommRing R₀] [Algebra R₀ R] [Algebra R₀ S] [IsScalarTower R₀ R S]
+
+open scoped nonZeroDivisors
+
+noncomputable section
+
+instance : Algebra (FractionRing R) (FractionRing S) := FractionRing.liftAlgebra _ _
```

Comment:
Aren't we in trouble when `S := FractionRing R`?

Reply:
I don't think we will ever see `FractionRing (FractionRing R)` in mathlb.

Reply:
Well, either we do need to worry and we should stick with the status quo, or we don't and `liftAlgebra` itself should be an instance, and have it's docstring adjusted to explain why we're not worried.

Reply:
`liftAlgebra` should not be instances for different reasons. If you have a `IntermediateField (FractionRing R) L` then you have two algebra instances on it.

Reply:
The docstring of [FractionRing.liftAlgebra](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Localization/FractionRing.html#FractionRing.liftAlgebra) explains that we get a diamond on `Algebra (FractionRing R) (FractionRing R)` if it's made an instance.